app/services/google/parser_service.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Google
+  class ParserService
+    TOP_ADS_SECTION_ID = 'tvcap'
+    BOTTOM_ADS_SECTION_ID = 'bottomads'
+    ADS_CLASS = 'uEierd'
+    SEARCH_SECTION_ID = 'search'
+    SEARCH_RESULT_CLASS = 'yuRUbf'
+
+    def initialize(html)
+      @html = html
+      @doc = Nokogiri::HTML(html)
```

Comment:
I would recommend also to move the Nokogiri call into the call method - to delay its initialization until the time we are sure it will be needed :)